extensions/wikia/FilePage/WikiaFilePage.php

```diff
@@ -40,29 +40,105 @@ protected function isVideo() {
 	/**
 	 * Render the image or video
 	 */
-	protected function openShowImage(){
+	protected function openShowImage() {
 		if ( $this->isVideo() ) {
 			$this->openShowVideo();
 		} else {
 			parent::openShowImage();
 		}
 	}
 
+	/**
+	 * Render the image or video
+	 */
+	public function view() {
+		global $wgMemc, $wgDBprefix;
+		if ( !$this->getContext()->getUser()->isAnon() ) {
+			parent::view();
+
+			return;
+		}
+		//fallback to main page
+		$url = Title::newMainPage()->getFullURL();
+		//wiki needs read privileges
+		if ( !$this->getTitle()->userCan( 'read' ) ) {
+			$this->getContext()->getOutput()->redirect( $url );
+
+			return;
+		}
+		$redirKey = wfMemcKey( "redir:", $wgDBprefix, $this->getTitle()->getPrefixedText() );
```

Comment:
`$wgDBprefix` is unneeded, since `wfMemcKey` will prefix the memcache key with `$wgDBname-$wgDBprefix`. `"redir:"` can also just be `'redir'` since it joins the parameters with a colon.